### DISCUSSION TITLE
Ezra Bible App 1.13

### DISCUSSION THREAD

**Discussion by tobias-klein** [ezra-bible-app/ezra-bible-app#1014] [General]
**Ezra Bible App 1.13** contains various bugfixes and enhancements and it comes with the newly translated Slovenian locale.

Note-worthy enhancements and fixes:
* Add Slovenian locale, thanks to Marjan Šavli. ([#1011][i1011])
* Add support for selecting verses based on pressing Shift key. ([#957][i957])
* Add a short cut for selecting all verses in the search results or a tagged verse list (CTRL + A). ([#959][i959])
* Copy verses to clipboard including html formatting. ([#962][i962])
* Dictionary use without keyboard. ([#912][i912])
* Put transChange meta data in html title attribute to make it readable. ([#972][i972])
* Add buttons for copying verses from translation comparison panel to clipboard. ([#537][i537])
* Add dialog that asks for confirming privacy options at first start of the app. ([#991][i991])
* Add ability to switch tag group within tag statistics panel. ([#995][i995])
* Fix: Greek Strong's parsing broken with StrongsGreek version 2.0. ([NSI #49][nsi49])
* Fix: Verse references are sometimes not parsed correctly when tagging verses. ([#979][i979])
* Fix: Saving current tab configuration not working on Android exit, but only on pause. ([#997][i997])

A detailed listing of all changes can be found in the [changelog](https://github.com/ezra-bible-app/ezra-bible-app/blob/master/CHANGELOG.md).

Installation instructions can be found [here](https://ezrabibleapp.net/installation).

[i537]: https://github.com/ezra-bible-app/ezra-bible-app/issues/537
[i912]: https://github.com/ezra-bible-app/ezra-bible-app/issues/912
[i957]: https://github.com/ezra-bible-app/ezra-bible-app/issues/957
[i959]: https://github.com/ezra-bible-app/ezra-bible-app/issues/959
[i962]: https://github.com/ezra-bible-app/ezra-bible-app/issues/962
[i972]: https://github.com/ezra-bible-app/ezra-bible-app/issues/972
[i979]: https://github.com/ezra-bible-app/ezra-bible-app/issues/979
[i991]: https://github.com/ezra-bible-app/ezra-bible-app/issues/991
[i995]: https://github.com/ezra-bible-app/ezra-bible-app/issues/995
[i997]: https://github.com/ezra-bible-app/ezra-bible-app/issues/997
[i1011]: https://github.com/ezra-bible-app/ezra-bible-app/pull/1011
[nsi49]: https://github.com/ezra-bible-app/node-sword-interface/issues/49

<hr /><em>This discussion was created from the release <a href='https://github.com/ezra-bible-app/ezra-bible-app/releases/tag/'>Ezra Bible App 1.13</a>.</em>

**Comment by tobias-klein** [comment]
The macOS build will follow tomorrow ... Apple has disabled my account and I am waiting for them to enable it again (that apparently takes 24h). I cannot notarize the app before my Apple account has been enabled again.

**Comment by Berovowilare** [comment]
Thanks.

**Comment by tobias-klein** [comment]
The macOS dmg file is now available! :)

https://github.com/ezra-bible-app/ezra-bible-app/releases/download/1.13.0/ezra-bible-app_mac_1.13.0_x64.dmg

**Comment by Mc-beat** [comment]
verse 13 causes problems.
On Big Sure (mac) I can see the books but cannot select the verses. They are not visible. Verse 12 works, but I miss copying information into the comment window.
What I also miss is the direct verse search with the keyboard and not just with the mouse.

Otherwise nice app

**Reply by tobias-klein** [reply]
@Mc-beat 
Thanks for providing feedback.

1) Regarding the issues with version 1.13, could you share a screenshot of the issue?
2) What exactly do you expect regarding "copying information into the comment window"?
3) Direct verse search with the keyboard, could you elaborate a bit more?

**Comment by ganakee** [comment]
I installed v.1.13 and have several Bibles installed. I am using an Ubuntu 22.04 variant.
The GUI loads but nothing ever displays on the right pane. I select the KJV Bible and then a book (e.g. Jonah). Nothing happens. I then try, with a different Bible selected, to Search. Nothing. There is just a three-dot indicator showing some activity but after minutes nothing displays. I cannot seem to get anything to display in the right-pane.

**Comment by ganakee** [comment]
I also uninstalled, rebooted, and reinstalled. I also deleted the ~/.sword directory, reboot, etc. Even after a fresh reinstall, you select a Bible and then a Book and nothing. (The Bibles etc. are installed using the Module Manager in the App.)

**Reply by Mc-beat** [reply]
This is exactly what I have on Mac. 
Exactly the same problem as you describe. I can't see anything.
Version 12 works reasonably well

**Comment by tobias-klein** [comment]
@Mc-beat @ganakee The issues you are describing sound like an essential component of Ezra Bible App (node-sword-interface) is not functioning.

Are you able to install modules via the install assistant in Ezra Bible App, using version 1.13?

I just tried to install Ezra Bible App 1.13 on Ubuntu 22.04 and it worked, so I could not reproduce the issue so far.

I will give it a try on MacOS, too - but that's for another day.

**Reply by Mc-beat** [reply]
Version 13 doesn't work for me on Mac Big Sure. Version 12 is running but some windows cannot be clicked. 

I have now switched to a different software. Very unfortunate. Maybe version 14 will run?

It really bothered me that you can't enter a verse directly, but always have to look for the books. ;(

**Reply by tobias-klein** [reply]
@Mc-beat The new release [1.14](https://github.com/ezra-bible-app/ezra-bible-app/releases/tag/1.14.0) is available since today.

**Comment by ganakee** [comment]
Thank you @tobias-klein.
I apologize for the length but try to provide some troubleshooting. At the end, 1.13 started working again BUT NOT AFTER REBOOT. However, there were multiple steps to get it working. Something seems to be wrong with 1.13 on Ubuntu--might work initially but not later. See my second post with details on after reboot.

### Used App to Install Bibles
Yes, I installed Bibles and related materials via the in-app module manager .

### First Run of 1.13 Was OK
I should clarify. I did an initial install of 1.13 and the app worked fine--at first. I did: 

1. Install Ezra via Eddy, 
2. use the in-app module manager to install some Bibles--Greek, Hebrew, and English language along with basic tools like the dictionaries. 
3. This worked.

### Problems Started Launching Later After Boot
I shutdown and came back about two days later. That is when, for me, the "nothing" issue started. I saw nothing in the right pane ever again.

Afterwards, to troubleshoot, I uninstalled Ezra using Eddy, rebooted, removed the `~/.sword` directory, rebooted, and then reinstalled Ezra. I have done this twice. This did NOT work or correct the issue.

On reinstall, I can use the Module Manager to re-download the Bibles etc. Yet, nothing displays.

### Troubleshooting With 1.12
I went to the Git Site and downloaded 1.12. I installed using Eddy as before. Works immediately.

I tried searching for where Ubuntu installs Ezra to see if I can find a SWORD executable. For others, Ezra on Ubuntu installs to:
```
Desktop Icon = /usr/share/applications/ezra-bible-app.desktop
Executable = /usr/bin/ezra-bible-app 
/usr/lib/ezra-bible-app
```
To manually launch Ezra from a command line, `ezra-bible-app`

### 1.12 Runs From Command Line and Gnome Desktop
```
ezra-bible-app
Working with userDataPath /home/someone/.config/ezra-bible-app
Using settings file /home/someone/.config/ezra-bible-app/config.json
[10870:0619/211747.590185:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
Using default database dir for database access!
Initializing database at /home/someone/.config/ezra-bible-app
Working with userDataPath /home/someone/.config/ezra-bible-app
Using settings file /home/someone/.config/ezra-bible-app/html-cache.json
Startup: 2.348s
[10829:0619/211749.941060:ERROR:nss_util.cc(286)] After loading Root Certs, loaded==false: NSS error code: -8018
Log from renderer: Persisting data
Log from renderer: Saving tab configuration
Log from renderer: Saving tag group id
IpcSettingsHandler: Setting lastUsedTagGroupId to null
Log from renderer: Saving last locale
IpcSettingsHandler: Setting cacheLocale to en
Log from renderer: Saving last used version
IpcSettingsHandler: Setting lastUsedVersion to 1.12.0
IpcSettingsHandler: Setting tabConfigurationTimestamp to Wed Jun 19 2024 21:17:51 GMT-0400 (Eastern Daylight Time)
```
### Second round Troubleshooting 1.13 By Installing as Update Via Eddy From 1.12

1. Install 1.12 using Eddy.
2. Open Eddy and display Eddy-installed apps.
4. Using Eddy, select UPDATE from within Eddy.
5. Updates to 1.13.
6. Use command line to run Ezra: 
7. ezra-bible-app

```
Working with userDataPath /home/someone/.config/ezra-bible-app
Using settings file /home/someone/.config/ezra-bible-app/config.json
Not configuring Sentry based on opt-out.
[12161:0619/212852.000367:ERROR:web_contents_preferences.cc(261)] preload script must have absolute path.
[12196:0619/212852.138142:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
Using default database dir for database access!
Initializing database at /home/someone/.config/ezra-bible-app
Working with userDataPath /home/someone/.config/ezra-bible-app
Using settings file /home/someone/.config/ezra-bible-app/html-cache.json
IpcSettingsHandler: Setting lastUsedTagGroupId to null
Startup: 2.289s
IpcSettingsHandler: Setting lastUsedVersion to 1.13.0
```


I am running POP OS 22.04 with all updates.

**Comment by ganakee** [comment]
I wanted to re-test after a reboot.
### 1.13 Does NOT Work After Reboot
After reboot, 1.13 does **NOT** work (even though it did above).

1. Install 1.13 as above.
2. Open a book--I used Jonah.
3. Jonah displays.
4. Close Ezra.
5. Reboot.
6. Launch Ezra.
7. **Nothing issue starts.**
8. I can see part of the text from the Book selected before (this just seems residual). However, I cannot select another book. I cannot open a new tab and select any books. You see the books which you can select but nothing happens when selected. Sometimes they highlight but sometimes they do not. Nothing shows in the pane.

Launching from command line:
```
ezra-bible-app
Working with userDataPath /home/someone/.config/ezra-bible-app
Using settings file /home/someone/.config/ezra-bible-app/config.json
Not configuring Sentry based on opt-out.
[4079:0619/215631.354704:ERROR:web_contents_preferences.cc(261)] preload script must have absolute path.
[4112:0619/215631.463939:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
Using default database dir for database access!
Initializing database at /home/someone/.config/ezra-bible-app
Working with userDataPath /home/someone/.config/ezra-bible-app
Using settings file /home/someone/.config/ezra-bible-app/html-cache.json
IpcSettingsHandler: Setting lastUsedTagGroupId to null
Startup: 2.513s
IpcSettingsHandler: Setting lastUsedVersion to 1.13.0
```
**Nothing displays in right-pane.**

**Reply by tobias-klein** [reply]
@ganakee 
Thank you so much for taking the time to troubleshoot, I appreciate it. It is still hard to understand the root cause of this bug. I can do some more testing soon and also analyze the differences between 1.12 and 1.13 that could potentially cause such a bug.

I see from the startup log above that in 1.13 you opted out for sending error reports.
Could you turn on that option (2nd option under *General options*) and then try again?
What this does is automatically submit debug information to the [Sentry](https://sentry.io/) Debug platform in case a JavaScript error occurs. The debug info does not contain any user data, but only internal trace information that helps to understand where JavaScript issues originate.
If I see a stack trace on Sentry based on the error happening on your machine then it would be easier for me to understand where this error is coming from in the code.

**Reply by tobias-klein** [reply]
@ganakee
So, from all the changes I implemented in 1.13 the only change that may have caused such an issue is the one implemented in the context of #997.
Once the error has occurred, could you close the app, then delete the file `/home/someone/.config/ezra-bible-app/html-cache.json` and then start the app again?
That file contains a cashed version of the lastly opened tabs. If something goes wrong with the persistence I could imagine that you may end up with an empty panel, as you have described it.

**Comment by ganakee** [comment]
Thank you @tobias-klein.

### Possible Clues and Exact Steps

1. I started the laptop with Ezra.
2. Booted and logged in.
3. I saw your two email responses.
4. I opened the Ezra App from the GNOME Desktop Icon.
5. I tried to open a book...nothing.
6. **Now, I did exactly these steps:** 
7. I clicked the `Gear (Options) Icon`.
8. Scrolled to `General Options`.
9. Selected `Send Error Reports` (This checkbox was OFF/empty, and I turned it ON/checked).
10. The `Gear (Options) Pane `disappeared.
11. **Immediately, a second tab that previously was trying to open Jonah from the KJV populated.**

I tried a few books and a search. **All worked.** I also tried some of the Commentaries and that worked along with left-pane parallel passages.
I also rebooted and then re-launched Ezra. **Ezra worked.**

I will try to break things by turning off Error Reporting and then rebooting....

**Reply by ganakee** [reply]
**Deliberate Trying to Break Things**
After the above and with Ezra open:

1. I clicked the Gear (Options) Icon.
2. Scrolled to General Options.
3. UN-selected Send Error Reports (checkbox now OFF/empty).
4. The Gear (Options) Pane disappeared.

I could not get any books to load. The "nothing" issue started again. The only user change was turning off send error reports.

After the above and with Ezra still open:
1. I clicked the Gear (Options) Icon.
2. Scrolled to General Options.
3. Again, I selected Send Error Reports (checkbox ON).
4. The Gear (Options) Pane disappeared.
5. This time, books would not load--nothing issue persisted.
6. I closed Ezra and relaunched Ezra.
7. Nothing. Cannot access books.
8. I rebooted.
9. I re-opened Ezra.
10. **Ezra now working/loading books.** (Note: Send Error Reports is ON.)

I don't want to give any red-herrings, but my limited testing may suggest that the **nothing-displayed issue may be linked to Send Error Reports**.

I do not have more time to night but hope to try deleting the config file as requested tomorrow.

**Reply by tobias-klein** [reply]
@ganakee Thanks for further testing! We're on to something!
When playing with the *Send Error Reports* option I found some JavaScript errors with the production version of 1.13. Those errors will be fixed in the next release, which should happen in the next few weeks. If you are curious: The root-cause are some uninitialized variables in case the error reporting is *turned off*. This means that turning the error reporting on right after starting the app and then once more restarting the app should be a workaround.

Would you be interested in testing a beta based on the bugfixes I perform?

**Reply by Mc-beat** [reply]
Okay. Now it's running. I have to delete everything, even in the library. Really everything and then reinstall ;)

**Reply by Mc-beat** [reply]
> @ganakee Thanks for further testing! We're on to something! When playing with the _Send Error Reports_ option I found some JavaScript errors with the production version of 1.13. Those errors will be fixed in the next release, which should happen in the next few weeks. If you are curious: The root-cause are some uninitialized variables in case the error reporting is _turned off_. This means that turning the error reporting on right after starting the app and then once more restarting the app should be a workaround.
> 
> Would you be interested in testing a beta based on the bugfixes I perform?

Many modules are now gray and I can not remove or update them, even if I delete them from the library it does not work, also no download and insert. 
Hmm.

**Reply by tobias-klein** [reply]
The issue regarding modules grayed out is completely separate from the discussion above.
Here the reason is that there are different SWORD paths used by different programs on macOS. While modules may be used by Ezra, they may not be deleted or modified if they are stored in a SWORD path that is not actively managed by Ezra. What could be done is to improve the transparency of how this information is reported.

The path actively managed by Ezra on macOS is currently ~/.sword, while other applications are using  ~/Library/Application Support/Sword.

Something we may do in the future is to standardize the path, stop using ~/.sword and use the same path that other applications are using.

That's related to this issue on node-sword-interface, the C++ library that manages this aspect.
https://github.com/ezra-bible-app/node-sword-interface/issues/41

If you want to clean this up, you should manually move everything from ~/Library/Application Support/Sword to ~/.sword.

**Comment by ganakee** [comment]
Yes @tobias-klein. I am willing to test a beta.

### Just Notes
Thanks for looking into this. Glad that you have a lead on the issue. No problem with waiting for the 1.14 release for a more permanent fix. (And thanks for your hard work on Ezra overall.)

I also appreciate the new 1.13 ability to turn off telemetry/error reports. Glad that this can still be an option  especially for those in at-risk environments where even anonymous telemetry could have consequences. (Now that I know what the send-error-reports is used for, I plan to leave it on to help in my personal case.)

Thanks again.

**Reply by ganakee** [reply]
@tobias-klein 
I downloaded (2024-06-21) from Git and built on POP OS/Ubuntu 22.04 the 1.14.0-dev. For others, see [https://github.com/ezra-bible-app/ezra-bible-app.git](https://github.com/ezra-bible-app/ezra-bible-app.git).

I used your instructions from `BUILD.MD` for Linux--for others, see [https://github.com/ezra-bible-app/ezra-bible-app/blob/master/BUILD.md](https://github.com/ezra-bible-app/ezra-bible-app/blob/master/BUILD.md). **If you wish me to run a build that you do, just let me know; or if you have a different GIT URL for a different version, I can try that if you give the exact URL.**

I run Ezra from `npm start`. The command line shows (just for reference):
```
npm start

> ezra-bible-app@1.14.0-Dev start
> electron .

Working with userDataPath /home/someone/.config/ezra-bible-app
Using settings file /home/someone/.config/ezra-bible-app/config.json
[14118:0621/110201.485082:ERROR:web_contents_preferences.cc(261)] preload script must have absolute path.
Using default database dir /home/someone/.config/ezra-bible-app for database access!
Initializing database at /homesomeone/.config/ezra-bible-app
Working with userDataPath /home/someone/.config/ezra-bible-app
Using settings file /home/someone/.config/ezra-bible-app/html-cache.json
Startup: 1.708s
IpcSettingsHandler: Setting lastUsedVersion to 1.14.0-Dev
[14118:0621/110203.299341:ERROR:nss_util.cc(286)] After loading Root Certs, loaded==false: NSS error code: -8018
IpcSettingsHandler: Setting bibleTranslation to BSBdraft
IpcSettingsHandler: Setting tabConfigurationTimestamp to Fri Jun 21 2024 11:02:13 GMT-0400 (Eastern Daylight Time)
```
### Preliminary Testing
I have been testing with turning `Send-Error-Reports` on and off multiple times. So far, Ezra seems to continue to work fine and books are displayed. I hope to do more testing later.

### Caveats/Limits

1. The 1.14-dev was built on a DIFFERENT computer system from the computer system cited above. Thus, for example, the ~/.sword directory would be for _this_ machine, not the one used before.
2. I followed your build instructions precisely. However, this is built on my system so some dependencies/versions may vary. The dependency versions would be the current versions for the 22.04 release so may lag.
3. I have only run Ezra within the project directory via `npm start`. Thus, I did NOT do a `release directory model.` (But might  if this helps you.)
4. The testing was limited and I did NOT do a reboot between. However, in 1.13, things immediately broke on toggling Send-Error-Reports, but I am not seeing that so far in 1.14-dev.
5. I did all from the command line but can view the project and import the GIT project in [VS Codium](https://vscodium.com/) (the telemetry-less Microsoft VS Code).

**Reply by tobias-klein** [reply]
Excellent. :)
So you ended up building your own snapshot build. That worked out well, congrats!
Based on your feedback, it seems that the fixes implemented based on #1048 are fixing the issue we have been discussing here. That's great!

I'll wait for some more feedback before closing that issue.

1.14 should be released soon. I am waiting for updates from the translators still. But otherwise I consider the scope complete (other open items will move to the next release).

**Reply by tobias-klein** [reply]
@ganakee I was a little too quick in my response. This particular error pattern can only appear based on a fully packaged (production version) of Ezra. You can get that package by additionally running `npm run package-linux` after running your build. The reason is that the error reporting is completely turned off in development builds (in order to avoid spamming the Sentry platform during development). That was also the reason why this error "slipped through" when I released 1.13.

**Comment by ganakee** [comment]
@tobias-klein 
OK. I will need to push this until tomorrow.
I tried running `npm run package-linux`  but got a dependency error. I will need to research this tomorrow. It looks like my default 22.04 packages are too old.

```
npm run package-linux 

> ezra-bible-app@1.14.0-Dev package-linux
> electron-packager . ezra-bible-app --overwrite --platform=linux --arch=x64 --prune=true --out=release --electron-version=17.1.0 --icon=icons/ezra.png

CANNOT RUN WITH NODE 12.22.9
Electron Packager requires Node >= 14.17.5.
```

**Comment by ganakee** [comment]
@tobias-klein Glad to help.

### Short Response
With the 2024-06-21 snapshot of 1.14-dev, I cannot get the `npm run package-linux` to compile:
```
npm run package-linux

> ezra-bible-app@1.14.0-Dev package-linux
> electron-packager . ezra-bible-app --overwrite --platform=linux --arch=x64 --prune=true --out=release --electron-version=17.1.0 --icon=icons/ezra.png

Packaging app for platform linux x64 using electron v17.1.0
ENOENT: no such file or directory, stat 'node_modules/node-sword-interface/sword/bindings/Android/SWORD/app/src/main/cpp/biblesync'

```
I tried this twice and it fails.

### Longer Response

1. Using the original machine, I downloaded a fresh copy of the 1.14-dev snapshot.
2. I updated my nodejs to `node -v v22.3.0` (Instructions at [https://nodejs.org/en/download/package-manager](https://nodejs.org/en/download/package-manager)
3. Within the git snapshot downloaded directory, `npm install`. Warnings but no errors.
4. Ran` npm run rebuild`. Deprecation warning but no errors.
5. Ran `npm start.` Ezra opens fine and all books working.
6. Ran` npm run package-linux`. **Error and no output directory created.**

**Error:**
```
npm run package-linux

> ezra-bible-app@1.14.0-Dev package-linux
> electron-packager . ezra-bible-app --overwrite --platform=linux --arch=x64 --prune=true --out=release --electron-version=17.1.0 --icon=icons/ezra.png

Packaging app for platform linux x64 using electron v17.1.0
ENOENT: no such file or directory, stat 'node_modules/node-sword-interface/sword/bindings/Android/SWORD/app/src/main/cpp/biblesync'

```

I checked `node_modules/node-sword-interface/sword/bindings/Android/SWORD/app/src/main/cpp/biblesync`  for me that is at:
`~/Programming/JavaScript/Ezra/ezra-bible-app-master/node_modules/node-sword-interface/sword/bindings/Android/SWORD/app/src/main/cpp`
There is a `biblesync` file there of 38bytes but it shows as a **symbolic link**.
I used `ls -la` to see where the symbolic link goes and see 
`lrwxrwxrwx 1 someone someone   38 Jun 21 21:00 biblesync -> ../../../../../../../../biblesync/src/`

I tried to go to that directory in terminal. :
```
~/Programming/JavaScript/Ezra/ezra-bible-app-master/node_modules/node-sword-interface/sword/bindings/Android/SWORD/app/src/main/cpp$ 
cd  ../../../../../../../../biblesync/src/c/
bash: cd: ../../../../../../../../biblesync/src/: No such file or directory

```
I also tried to manually navigate up but am not seeing the linked materials.

UPDATE:
I also used `cd ../../../../../../../../` and then looked for `biblesync`. I do not see it. `ls` output:

```
~/Programming/JavaScript/Ezra/ezra-bible-app-master/node_modules/node-sword-interface$ ls
API.md       build           deploy_local.sh  locales.d     patch      src
bin          CMakeLists.txt  examples         node_modules  README.md  sword
binding.gyp  COPYING         index.js         package.json  scripts    sword_build
```

**Reply by tobias-klein** [reply]
@ganakee

You're going through a lot of trouble here. Thanks for taking the time.

I am aware of this issue with the packaging connected to the files in node-sword-interface/sword/**.

Please execute the step `npm run purge-build-artifacts` before you run `npm run package-linux`.
That should solve your issue above.

Whatever the issue is with that strange symlink, etc. The solution is to delete all build output previously generated in the context of building node-sword-interface (that automatically happens as part of `npm run rebuild`) - before proceeding with the packaging.

I will update the build instructions in a moment to reflect that.

**Reply by tobias-klein** [reply]
I have just updated the [build instructions](https://github.com/ezra-bible-app/ezra-bible-app/blob/master/BUILD.md#create-a-release-directory-1) for Linux and macOS - adding that missing step mentioned above.

**Comment by ganakee** [comment]
@tobias-klein 
Thank you. Not a problem and happy to help. This is a good project. (And I have not used **npm** so a learning experience.)

**The code changes seem to correct the issue** of books not showing when the `Send Error Reports `checkbox status is changed in Ezra App 1.13.

### Pre-testing Build Successful

1. I re-did all steps for building Ezra App.
2. `npm install`
3. `npm run rebuild`
4. `npm run purge-build-artifacts`
5. `npm run package-linux`
6. From terminal, `cd` to the release directory. `release/ezra-bible-app-linux-x64`
7. Verified ezra executable exists.
8. Ran ./`ezra-bible-app`
9. Verified 1.14-dev in Ezra App title bar.

### Testing Successful
I ran multiple tests. I toggled Send Error Reports on/off going back and forth and opening books etc in between (trying as many combinations as I could).

1.  Open books from multiple Bible versions
2. Close books
3. Close all tabs and reopen a tabs
4. Try to open Dictionaries and Commentaries for selected verses.
5. Close Ezra/ Open Ezra
6. Also combinations of on/off and rebooting in-between.

In all cases, 1.14-dev performed flawlessly with my local build of 1.14-dev (production version in `release/ezra-bible-app-linux-x64`).

**Reply by tobias-klein** [reply]
Thank you for all the time spent on testing, @ganakee! Release will be next weekend once all the local updates are in.
